src/main/java/br/com/meli/exercises/repositories/StarWarsRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package br.com.meli.exercises.repositories;
+
+import br.com.meli.exercises.model.Character;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.ResourceUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class StarWarsRepositoryImpl implements StarWarsRepository{
+
+    private boolean matchWith(String query, Character character) {
+        return character.getName().toUpperCase().contains(query.toUpperCase());
+    }
+
+    @Override
+    public List<Character> findCharacteresByName(String nameCharacter) {
```

Comment:
```suggestion
    public List<Character> findCharacteresByNameContaining(String nameCharacter) {
```

só para deixar mais claro o que o método está fazendo 😉